frontend/src/Components/Endpoint/Endpoint.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import Button from "plaid-threads/Button";
+import Note from "plaid-threads/Note";
+
+import Table from "../Table/Table";
+import Error from "../Error/Error";
+
+import {
+  DataItem,
+  Categories,
+  ErrorDataItem,
+  Data,
+} from "../../Utilities/dataUtilities";
+
+import styles from "./Endpoint.module.scss";
+
+interface Props {
+  endpoint: string;
+  name?: string | null;
+  categories: Array<Categories>;
+  schema: string;
+  description: string;
+  transformData: (arg: any) => Array<DataItem>;
+}
+
+const Endpoint = (props: Props) => {
+  const [showTable, setShowTable] = useState(false);
+  const [transformedData, setTransformedData] = useState<Data>([]);
+  const [isError, setIsError] = useState(false);
+  const [error, setError] = useState<ErrorDataItem>({
+    error_type: "",
+    error_code: "",
+    error_message: "",
+    display_message: "",
+    status_code: null,
+  });
+
+  const getData = async () => {
+    const response = await fetch(`/api/${props.endpoint}`, { method: "GET" });
+    const data = await response.json();
+    if (data.error != null) {
+      setError(data.error);
+      setIsError(true);
+    } else {
+      setTransformedData(props.transformData(data)); // put data in propeer format for each individual product
```

Comment:
(nit) 
```suggestion
      setTransformedData(props.transformData(data)); // transform data into proper format for each individual product
```

Reply:
this could be an early return instead of an else

Reply:
done

Reply:
i'm confused...which lines are you referring to here?

Reply:
oh...i see.